### PR TITLE
FIPS:add git test case under the fips environment

### PR DESCRIPTION
**Pull request by dehai**
This script based on test case #1525281-FIPS: git, should be running at
fips environment, to check whether can be work with ssh or https protocol.

Local verification run:http://147.2.212.179/tests/366

### CODE REVIEW

File: tests/fips/git_fips.pm [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+
+sub run() {
+
+    my $password = $testapi::password;
+    my $username = $testapi::username;
+    my $email    = "you\@example.com";
+    select_console "root-console";
+
+    # Create a test repo
+    assert_script_run("zypper -n in git-core", 190);
```
Comment (coolo) [inline tests/fips/git_fips.pm]: better use zypper_call

File: tests/fips/git_fips.pm [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+    assert_script_run("git config --global user.name \"$username\"");
+    assert_script_run("git add README;git commit -m \"Initial commit\"");
+
+    # Clone to a bare git repo
+    assert_script_run("cd ~/repos;git clone --bare qa1 qa0");
+
+    # Clone repo via ssh
+    type_string("clear\n");
+    script_run("git clone ssh://localhost:/root/repos/qa0 qa2 | tee /dev/$serialdev", 0);
+    if (assert_screen("input-yes", 60)) {
```
Comment (coolo) [inline tests/fips/git_fips.pm]: assert_screen with if doesn't make sense. assert_screen returns the match or dies. There is no else

File: products/sle/main.pm [hunk 3]
```diff
@@ -914,6 +914,7 @@ sub load_fips_tests_misc() {
     loadtest "console/journald_fss.pm";
     loadtest "fips/curl_fips_rc4_seed.pm";
     loadtest "console/consoletest_finish.pm";
+    loadtest "fips/git_fips.pm";
```
Comment (coolo) [inline products/sle/main.pm]: as there is nothing fips specific in your test, can you just make that a console test and also load it in opensuse and sle's extra test?

File: tests/fips/git_fips.pm (hunk 1 above)
Comment (dehai) [inline tests/fips/git_fips.pm]: ok, I will switch to it .

**Comment by dehai** [comment]
Local verification run:http://147.2.212.179/tests/371

File: products/sle/main.pm [hunk 4]
```diff
@@ -914,6 +914,7 @@ sub load_fips_tests_misc() {
     loadtest "console/journald_fss.pm";
     loadtest "fips/curl_fips_rc4_seed.pm";
     loadtest "console/consoletest_finish.pm";
+    loadtest "console/git_fips.pm";
```
Comment (coolo) [inline products/sle/main.pm]: again, I don't see this as fips specific. I understand you wrote it with fips in mind and it should run in the fips_tests. But the test itself just tests git and nothing about fips.

So please just name it git and load the test in opensuse/main.pm and sle/main.pm as console test.

**Comment by dehai** [comment]
@coolo based on your comments, I rename this script to git .pm and add this into console directory. modify opensuse/main.pm and sle/main.pm ,to add this into extra_test. If still something I haven't done, please let know.. thanks!!

**Comment by DimStar77** [comment]
Test fails on Tumbleweed: https://openqa.opensuse.org/tests/287774#step/git/20

The test relies on sshd being up and running but does not take care of enabling it; at the point it was added in extra_tests, there is no sshd running as it seems

**Comment by coolo** [comment]
I will take care

**Pull request merged**